src/kvirc/ui/KviInputEditor.cpp

```diff
@@ -2282,54 +2293,38 @@ void KviInputEditor::completion(bool bShift)
 				}
 			}
 
-			if(bInCommand && !bIsCommand)
-			{
+			if(bInCommand && !bIsCommand && !bIsFunction)
 				completionEscapeUnsafeToken(szMatch); // escape crazy things like Nick\nquit
-			}
 		}
 		else
 		{
 			QString szAll;
 			szMatch = tmp.front();
-			int iWLen = szWord.length();
-			if(szMatch.left(1) == '$')
-				szMatch.remove(0, 1);
-			for(auto szTmpIter : tmp)
+			auto predicate = bIsDir
```

Comment:
Can't we just put the `bIsDir` inside the lambda? That removes the need for `std::function`, the ternary stuff, and it just makes the code look cleaner overall.